datafusion-postgres/src/handlers.rs

```diff
@@ -50,72 +57,360 @@ impl PgWireServerHandlers for HandlerFactory {
+        let var_name = variable
+            .first()
+            .map(|ident| ident.to_string().to_lowercase())
+            .unwrap_or_default();
+
+        let sc_guard = self.session_context.read().await;
+        let config = sc_guard.state().config().options().clone();
+
+        let value = match var_name.as_str() {
+            "timezone" => config
+                .execution
+                .time_zone
+                .clone()
+                .unwrap_or_else(|| "UTC".to_string()),
+            "client_encoding" => self
+                .custom_session_vars
+                .read()
+                .await
+                .get(&var_name)
+                .cloned()
+                .unwrap_or_else(|| "UTF8".to_string()),
+            "search_path" => self
+                .custom_session_vars
+                .read()
+                .await
+                .get(&var_name)
+                .cloned()
+                .unwrap_or_else(|| "public".to_string()),
+            "application_name" => self
+                .custom_session_vars
+                .read()
+                .await
+                .get(&var_name)
+                .cloned()
+                .unwrap_or_else(|| "".to_string()),
+            "datestyle" => self
+                .custom_session_vars
+                .read()
+                .await
+                .get(&var_name)
+                .cloned()
+                .unwrap_or_else(|| "ISO, MDY".to_string()),
+            "client_min_messages" => self
+                .custom_session_vars
+                .read()
+                .await
+                .get(&var_name)
+                .cloned()
+                .unwrap_or_else(|| "notice".to_string()),
+            "extra_float_digits" => self
+                .custom_session_vars
+                .read()
+                .await
+                .get(&var_name)
+                .cloned()
+                .unwrap_or_else(|| "3".to_string()),
+            "standard_conforming_strings" => self
+                .custom_session_vars
+                .read()
+                .await
+                .get(&var_name)
+                .cloned()
+                .unwrap_or_else(|| "on".to_string()),
+            "check_function_bodies" => self
+                .custom_session_vars
+                .read()
+                .await
+                .get(&var_name)
+                .cloned()
+                .unwrap_or_else(|| "off".to_string()),
+            "transaction_read_only" => self
+                .custom_session_vars
+                .read()
+                .await
+                .get(&var_name)
+                .cloned()
+                .unwrap_or_else(|| "off".to_string()),
+            "transaction_isolation" => self
+                .custom_session_vars
+                .read()
+                .await
+                .get(&var_name)
+                .cloned()
+                .unwrap_or_else(|| "read committed".to_string()),
```

Comment:
Can we extract this to a function like `self.session_var(&var_name, "read_committed").await`